engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/EnsureCompiledJob.scala

```diff
@@ -223,15 +214,17 @@ class EnsureCompiledJob(protected val files: Iterable[File])
     */
   private def compile(
     module: Module
-  )(implicit ctx: RuntimeContext): Module = {
+  )(implicit ctx: RuntimeContext): Either[Throwable, Module] = {
     val prevStage = module.getCompilationStage
-    module.compileScope(ctx.executionService.getContext).getModule
+    val compilationResult = Either.catchNonFatal {
```

Comment:
This is lacking. The compiler will only throw on fatal compiler errors (i.e. "I can't proceed with compilation, something has gone horribly wrong") and every such instance should be considered a compiler bug.

You probably also want to sent the results of `ErrorGathering` pass to the user – these are errors in user code, which our compiler was able to recover from – things like variable redefinitions, missing names etc. These are important to report, as something in the file is broken, even if most of it can still function properly. I.e. these errors are non-fatal (the module is still compiled and kind of working, but user's attention is required).